packages/blocks/src/components/drag-handle.ts

```diff
@@ -373,18 +386,38 @@ export class DragHandle extends LitElement {
   };
 
   private _onDragStart = (e: DragEvent) => {
-    if (e.dataTransfer && this._startModelState) {
-      e.dataTransfer.effectAllowed = 'move';
-      this._draggingElement = document.querySelector(
-        `[${BLOCK_ID_ATTR}="${this._startModelState.model.id}"]`
-      );
-      if (this._draggingElement) {
-        // hack: set opacity to 0.9 to remove dragging element's shadow
-        // maybe the dragging element also has opacity?
-        this._draggingElement.style.opacity = '0.9';
-        e.dataTransfer.setDragImage(this._draggingElement, 0, 0);
-      }
+    const clickDragState = this._getBlockEditingStateByPosition?.(
+      this.getDropAllowedBlocks(null),
+      e.pageX,
+      e.pageY,
+      true
+    );
+
+    if (!clickDragState || !e.dataTransfer) {
+      return;
     }
+
+    e.dataTransfer.effectAllowed = 'move';
+
+    const selectedBlocks = this._getSelectedBlocks() ?? [];
+
+    // fixme: the block may not have block id?
+    const draggingBlockElements = selectedBlocks.includes(
+      clickDragState.element
+    )
+      ? selectedBlocks
+      : [clickDragState.element];
+
+    // hack: set opacity to 0.9 to remove dragging element's shadow
+    // maybe the dragging element also has opacity?
+    const firstElement = draggingBlockElements[0];
+
+    // fixme: handle multiple blocks case
+    // the drag image right now only renders the first block
```

Comment:
May need this https://stackoverflow.com/questions/34200023/drag-drop-set-custom-html-as-drag-image

Reply:
Let me have a try

Reply:
I tried a solution in [`14c7608` (#1346)](https://github.com/toeverything/blocksuite/pull/1346/commits/14c760883f7e47995e7fd8934deb512c19803ffb#diff-4e6ed695466ea09e7e5564e244dd064445d6379362b0bff473a400155f99ab94R479-R494).

It clones every selected item and then puts them into a new div element as the drag image. The drag image will be placed somewhere not to be seen by the users.
It does work at the beginning but I realize it has some side effects after I cloned the nodes. The nodes we are cloning are web components that will have the same lifecycles of the predefined behaviors and this will cause some unexpected issues.
Now the solution is to copy the selection rects instead when there are multiple selections as a workaround.

I also noticed that the solution does not work well in Safari, which is bad news as this issue will also happen in Tarui on Mac.